### PR TITLE
Bug 1809498: Fixes to support upgrade from 4.4 to 4.5

### PR DESCRIPTION
**Pull request by jcantrill**
This PR:

* reverts #341 as the change was too aggressive
* Checks container args when determining a change
* removes choosing the ES image from the CR
  * The image should not be independent of the operator that is proving config
  * ES CR provider could point to an ES5 image but operator only knows about ES6

Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1809498

Note:  Purposely has 3 commits to address three specific issues

### CODE REVIEW

**Comment by jcantrill** [comment]
may be blocked by https://github.com/openshift/origin-aggregated-logging/pull/1890

File: pkg/controller/elasticsearch/controller.go [hunk 1]
```diff
@@ -85,6 +87,29 @@ func (r *ReconcileElasticsearch) Reconcile(request reconcile.Request) (reconcile
...
+				break
+			}
+		}
+		if !exists {
+			cluster.Status.Conditions = append(cluster.Status.Conditions, loggingv1.ClusterCondition{
+				Type:               loggingv1.CustomImage,
+				Status:             v1.ConditionTrue,
+				LastTransitionTime: metav1.Now(),
+				Reason:             "CustomImageUnsupported",
+				Message:            "Specifiying a custom image from the customer resource is not supported",
```
Comment (syedriko) [inline pkg/controller/elasticsearch/controller.go:107]: custom resource?

**Pull request merged**